### ISSUE TITLE
Issues when training

### ISSUE THREAD

**Issue by Xiaosa1222** [JDSobek/MedYOLO#32]
Hello! I'm trying to implement plaque detection in CCTA images based on MedYolo, I set the size to 350, epoch to 1000, batch_size to 4, and finally I get the following results, it doesn't look like the results are good, I'd like to ask the author what's the reason for this? Is the target to be detected too small?

![Image](https://github.com/user-attachments/assets/98080861-f1d8-4110-80c6-dfc7848b73f1)
Meanwhile, the target cannot be detected when using detect.py.

![Image](https://github.com/user-attachments/assets/ea889e9d-5ee1-4b48-b176-baa9a4ff545f)

**Comment by JDSobek** [comment]
I think there's a decent chance plaques are too small, but I think your dataset is too small to tell. Since you only have 5 batches, a full 1000 epoch training run is only 5000 weight updates, many of which will be at fairly low learning rates if you're using the default one-cycle LR schedule, and that means there aren't many opportunities for your model to update its weights.

Is there a way for you to get more data or pre-train the model on another, larger dataset (ideally with a couple hundred examples)? If not, maybe you could try to heavily augment your existing dataset (by at least a factor of 10) or at greater risk of overfitting let it run for several thousand epochs. Since the models start randomly initialized, they need a good number of weight updates to start performing well.

**Comment by Xiaosa1222** [comment]
Thank you for your reply!
Actually, I have 200 training data (after rotated image enhancement) and 20 validation data, which means 50 batches for training and 5 batches for validation.
![Image](https://github.com/user-attachments/assets/a1efa184-8645-444c-9d68-5e7eaeecf05f)

**Comment by Xiaosa1222** [comment]
Should I add more data for training?

**Comment by JDSobek** [comment]
If you can add more data, that would be good.

How heavily augmented is your training set? It looks like it's optimizing on the training set pretty well, but if that's a relatively small set that's been heavily augmented, the model might not generalize to a validation set.

How well does the model do when you run validate.py or detect.py on the training set? That will tell us whether the model is having trouble identifying plaques or generalizing to new data.

**Comment by Xiaosa1222** [comment]
Thank you for your reply!
I augmented the data the way it is in your code and it should be augmented by a factor of six, meaning that one instance of the original data corresponds to six instances of the data used for training. My 200 training data should contain data from 34 patients.

![Image](https://github.com/user-attachments/assets/187f61f7-972f-4154-94a0-abdb6584345d)
I ran detect.py on 10 cases from the training set, and only six of the labels were recognized. The four plaques that were not recognized were very small, and the location of one of the recognized plaques didn't look right.

![Image](https://github.com/user-attachments/assets/e00b4a04-e127-438f-a514-f50c6be1ca00)

**Comment by JDSobek** [comment]
Ok, it sounds like it's learning but not generalizing so there's some hope, at least for the larger plaques. If you can get more training patients you'll probably see better results. The smallest training set I used was 60 patients, which I augmented 5x (so 360 total training examples) and tested against 20 validation patients.

It's also worth double checking whether your validation plaques are also very small. I'm not sure how to help the model detect very small plaques, I've been surprised with some results on small objects but other than the resizing to 350x350 I'm not sure what's all contributing to the model's struggles there.

TotalSegmentator has some annotated public datasets. I'm not sure if those datasets include the cardiovascular segmentations, but if they do you might be able to pretrain on those, and use those weights as the starting point for detecting plaques.

**Comment by Xiaosa1222** [comment]
Thank you for your reply!Thank you for your reply!
I will train the model on more data to see how it works.

**Comment by Xiaosa1222** [comment]
Thank you for your reply!
I will train the model on more data to see how it works.
